### ISSUE TITLE
Issue with 'minLength' and 'maxLength' property

### ISSUE THREAD

**Issue by andydor** [everit-org/json-schema#487]
```
"name": {
      "title": "Object name",
      "description": "Description of object name",
      "type": "string",
      "maxLength": 5
 }
```

When using the minLength or maxLength property, every time it leads to:

```
Exception in thread "main" org.everit.json.schema.SchemaException: #/properties/property2/maxLength: expected type: Integer, found: Long
	at org.everit.json.schema.loader.LoadingState.createSchemaException(LoadingState.java:151)
	at org.everit.json.schema.loader.JsonValue.requireInteger(JsonValue.java:235)
	at org.everit.json.schema.loader.JsonValue.requireInteger(JsonValue.java:228)
	at java.base/java.util.Optional.map(Optional.java:265)
	at org.everit.json.schema.loader.StringSchemaLoader.load(StringSchemaLoader.java:39)
	at org.everit.json.schema.loader.AbstractSchemaExtractor.buildStringSchema(SchemaExtractor.java:165)
	at org.everit.json.schema.loader.TypeBasedSchemaExtractor.loadForExplicitType(SchemaExtractor.java:271)
	at org.everit.json.schema.loader.JsonValue$Multiplexer.requireAny(JsonValue.java:47)
	at org.everit.json.schema.loader.TypeBasedSchemaExtractor.extract(SchemaExtractor.java:253)
	at org.everit.json.schema.loader.AbstractSchemaExtractor.extract(SchemaExtractor.java:114)
	at org.everit.json.schema.loader.SchemaLoader.runSchemaExtractors(SchemaLoader.java:424)
	at org.everit.json.schema.loader.SchemaLoader.loadSchemaObject(SchemaLoader.java:388)
	at org.everit.json.schema.loader.JsonValue$Multiplexer.requireAny(JsonValue.java:47)
	at org.everit.json.schema.loader.SchemaLoader.load(SchemaLoader.java:462)
	at org.everit.json.schema.loader.SchemaLoader.loadChild(SchemaLoader.java:466)
	at org.everit.json.schema.loader.ObjectSchemaLoader.addPropertySchemaDefinition(ObjectSchemaLoader.java:70)
	at org.everit.json.schema.loader.ObjectSchemaLoader.lambda$populatePropertySchemas$9(ObjectSchemaLoader.java:64)
	at org.everit.json.schema.loader.JsonObject.iterateOnEntry(JsonObject.java:84)
	at org.everit.json.schema.loader.JsonObject.lambda$forEach$1(JsonObject.java:79)
	at java.base/java.util.HashMap$EntrySet.forEach(HashMap.java:1039)
	at org.everit.json.schema.loader.JsonObject.forEach(JsonObject.java:79)
	at org.everit.json.schema.loader.ObjectSchemaLoader.populatePropertySchemas(ObjectSchemaLoader.java:61)
	at org.everit.json.schema.loader.ObjectSchemaLoader.lambda$load$0(ObjectSchemaLoader.java:32)
	at java.base/java.util.Optional.ifPresent(Optional.java:183)
	at org.everit.json.schema.loader.ObjectSchemaLoader.load(ObjectSchemaLoader.java:32)
	at org.everit.json.schema.loader.AbstractSchemaExtractor.buildObjectSchema(SchemaExtractor.java:139)
	at org.everit.json.schema.loader.TypeBasedSchemaExtractor.loadForExplicitType(SchemaExtractor.java:283)
	at org.everit.json.schema.loader.JsonValue$Multiplexer.requireAny(JsonValue.java:47)
	at org.everit.json.schema.loader.TypeBasedSchemaExtractor.extract(SchemaExtractor.java:253)
	at org.everit.json.schema.loader.AbstractSchemaExtractor.extract(SchemaExtractor.java:114)
	at org.everit.json.schema.loader.SchemaLoader.runSchemaExtractors(SchemaLoader.java:424)
	at org.everit.json.schema.loader.SchemaLoader.loadSchemaObject(SchemaLoader.java:388)
	at org.everit.json.schema.loader.JsonValue$Multiplexer.requireAny(JsonValue.java:47)
	at org.everit.json.schema.loader.SchemaLoader.load(SchemaLoader.java:462)
	at org.example.SchemaSampleValidator.validate(SchemaSampleValidator.java:45)
	at org.example.Main.main(Main.java:8)
```

The sample application can be found [here](https://github.com/andydor/json-schema.git)

**Comment by erosb** [comment]
Thank you for submitting a proper reproducer. Is [migration to the new library](https://github.com/erosb/json-sKema) suitable for you as a workaround?

**Comment by erosb** [comment]
Hello @andydor , it turns out this is an interoperatibility problem between GSON and JSON-Java. The `JSONObjectUtils.parse()` method you use parses the json number into a `Long` instance, but it isn't accepted by the `SchemaLoader`. I suggest skipping parsing the schema json into a `Map<String, Object> instance, and instead construct the `JSONObject` for the schema in the same way as you do for the input. 

TL;DR: change line 37 to `JSONObject rawSchema = new JSONObject(readFile(inputSchemaFilename));` and it works.

**Comment by andydor** [comment]
Thank you for your quick response and solution
